applications/maplab-server-node/test/test-maplab-server-node.cc

```diff
@@ -39,41 +45,46 @@ class MaplabServerNodeTest : public ::testing::Test {
 
 TEST_F(MaplabServerNodeTest, TestMaplabServerNode) {
   MaplabServerNode maplab_server_node;
+  maplab_server_node_ = &maplab_server_node;
   maplab_server_node.start();
   ros::Time::init();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap0));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap1));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap2));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap3));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap4));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap5));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap6));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
   EXPECT_TRUE(maplab_server_node.loadAndProcessSubmap(kRobotName, kSubmap7));
-  std::this_thread::sleep_for(std::chrono::seconds(1));
+  std::this_thread::sleep_for(std::chrono::seconds(5));
   maplab_server_node.visualizeMap();
 
+  while (getTotalNumMergedSubmaps() < kNumSubmaps) {
```

Comment:
added this since the test previously was kind of useless since the server would shutdown before the submaps are actually merged

Reply:
+1

Reply:
what do you mean with that?

Reply:
nice that you found that.